apps/internal/base/base.go

```diff
@@ -147,6 +147,15 @@ func WithCacheAccessor(ca cache.ExportReplace) Option {
 	}
 }
 
+// WithKnownAuthorityHosts specifies hosts Client shouldn't validate or request metadata for because they're known to the user
+func WithKnownAuthorityHosts(hosts []string) Option {
+	return func(c *Client) {
+		cp := make([]string, len(hosts))
+		copy(cp, hosts)
```

Comment:
go newbie question - why do you need to copy the hosts slice to cp? Is it because MSAL might append data to it?

Reply:
That's the gist. A slice value is a pointer. Appending copies a slice without modifying the original but elements are mutable otherwise. This explicit copying will prevent sharing memory with applications once this API is publicly accessible and `hosts` is provided by users.